$OpenBSD$

Hardcode TCP keepalive constants, skip CApiFFI on OpenBSD.
GHC 9.10 foreign import capi generates broken C code on OpenBSD.
Values from <netinet/tcp.h>: TCP_KEEPIDLE=0x100, TCP_KEEPINTVL=0x200, TCP_KEEPCNT=0x300.

--- src/Simplex/Messaging/Transport/KeepAlive.hs.orig
+++ src/Simplex/Messaging/Transport/KeepAlive.hs
@@ -50,12 +50,16 @@
 #else
 -- Mac/Linux

-#if defined(darwin_HOST_OS)
+#if defined(openbsd_HOST_OS)
+_TCP_KEEPIDLE :: CInt
+_TCP_KEEPIDLE = 0x100
+_TCP_KEEPINTVL :: CInt
+_TCP_KEEPINTVL = 0x200
+_TCP_KEEPCNT :: CInt
+_TCP_KEEPCNT = 0x300
+#elif defined(darwin_HOST_OS)
 foreign import capi "netinet/tcp.h value TCP_KEEPALIVE" _TCP_KEEPIDLE :: CInt
 #else
 foreign import capi "netinet/tcp.h value TCP_KEEPIDLE" _TCP_KEEPIDLE :: CInt
-#endif
-
-foreign import capi "netinet/tcp.h value TCP_KEEPINTVL" _TCP_KEEPINTVL :: CInt
-
-foreign import capi "netinet/tcp.h value TCP_KEEPCNT" _TCP_KEEPCNT :: CInt
+foreign import capi "netinet/tcp.h value TCP_KEEPINTVL" _TCP_KEEPINTVL :: CInt
+foreign import capi "netinet/tcp.h value TCP_KEEPCNT" _TCP_KEEPCNT :: CInt
+#endif
